script/package

```diff
@@ -4,7 +4,7 @@ set -e
 
 version=$(git rev-parse --short HEAD)
 
-sed -i '' -e"s/google_tracking_id:.*/google_tracking_id:/" _config.yml
+sed -i'' -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
```

Comment:
```suggestion
sed -i'' -e "s/tracking_id\s*:.*/tracking_id:/" _config.yml
```

Reply:
@bazzaar sorry, you're right that `\s` works. It wont work in shell (which is what this is executed as based on the top of the file) unless this space is present.

Reply:
```suggestion
sed -i "" 's/tracking_id\s*:.*/tracking_id:/' _config.yml
```

Ugh, sorry, @bazzaar. this is actually the change that is needed for it to work on my machine. Could you check if this works on yours? Basically the `-e` was causing my shell to continuously create a file, `_config.yml-e`.

Reply:
Hi Brent,

It's easy to get confused with all these different versions of the command, so this is what I did : 

I made a temp directory somewhere outside the project, and I put two copies of the `_config.yml` in there, renaming the 2nd one. I created a new `sed_script.sh` file in that temp dir also. Here's a listing : 

bazzaar@linux-box:~/git_work/temp> ls -1
_config.yml
_config.yml.bu
sed_script.sh

I made the `sed_script.sh` executable.

I edited the `sed_script.sh` file, and placed the following commands in it : 

```
#!/bin/sh

set -e
echo "------ -e option used"
echo "------"
echo "## 1 - Error \"sed: can't read : No such file or directory\""
echo "sed -i '' -e\"s/tracking_id\s*:.*/tracking_id:/\" _config.yml"
echo "test commented out"
#egrep 'tracking' _config.yml
#sed -i '''' -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
#egrep 'tracking' _config.yml
#cp _config.yml.bu _config.yml
echo "------"
echo "## 2 - ok"
echo "sed -i'' -e\"s/tracking_id\s*:.*/tracking_id:/\" _config.yml"
egrep 'tracking' _config.yml
sed -i'' -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
egrep 'tracking' _config.yml
cp _config.yml.bu _config.yml
echo "------"
echo "## 3 - ok"
echo "sed -i -e\"s/tracking_id\s*:.*/tracking_id:/\" _config.yml"
egrep 'tracking' _config.yml
sed -i -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
egrep 'tracking' _config.yml
cp _config.yml.bu _config.yml
echo "------"
echo "## 4 - ok"
echo "sed -i \"\" -e\"s/tracking_id\s*:.*/tracking_id:/\" _config.yml"
egrep 'tracking' _config.yml
sed -i -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
egrep 'tracking' _config.yml
cp _config.yml.bu _config.yml
echo " "
echo "------ -e option not used"
echo "------"
echo "## 5 - Error \"sed: can't read : No such file or directory\""
echo "sed -i '' 's/tracking_id\s*:.*/tracking_id:/' _config.yml"
echo "test commented out"
#egrep 'tracking' _config.yml
#sed -i '' 's/tracking_id\s*:.*/tracking_id:/' _config.yml
#egrep 'tracking' _config.yml
#cp _config.yml.bu _config.yml
echo "------"
echo "## 6 - ok"
echo "sed -i'' 's/tracking_id\s*:.*/tracking_id:/' _config.yml"
egrep 'tracking' _config.yml
sed -i'' 's/tracking_id\s*:.*/tracking_id:/' _config.yml
egrep 'tracking' _config.yml
cp _config.yml.bu _config.yml
echo "------"
echo "## 7 - ok"
echo "sed -i 's/tracking_id\s*:.*/tracking_id:/' _config.yml"
egrep 'tracking' _config.yml
sed -i 's/tracking_id\s*:.*/tracking_id:/' _config.yml
egrep 'tracking' _config.yml
cp _config.yml.bu _config.yml
echo "------"
echo "## 8 - ok"
echo "sed -i \"\" 's/tracking_id\s*:.*/tracking_id:/' _config.yml"
egrep 'tracking' _config.yml
sed -i 's/tracking_id\s*:.*/tracking_id:/' _config.yml
egrep 'tracking' _config.yml
cp _config.yml.bu _config.yml
```
Then I ran it, from within the temp dir, and here's the output : 

```
bazzaar@linux-box:~/git_work/temp> ./sed_script.sh
------ -e option used
------
## 1 - Error "sed: can't read : No such file or directory"
sed -i '' -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
commented out
------
## 2 - ok
sed -i'' -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
    tracking_id          : UA-3769691-2                               <----- before
    tracking_id:                                                                 <----- after
------
## 3 - ok
sed -i -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
    tracking_id          : UA-3769691-2
    tracking_id:
------
## 4 - ok
sed -i "" -e"s/tracking_id\s*:.*/tracking_id:/" _config.yml
    tracking_id          : UA-3769691-2
    tracking_id:
 
------ -e option not used
------
## 5 - Error "sed: can't read : No such file or directory"
sed -i '' 's/tracking_id\s*:.*/tracking_id:/' _config.yml
commented out
------
## 6 - ok
sed -i'' 's/tracking_id\s*:.*/tracking_id:/' _config.yml
    tracking_id          : UA-3769691-2
    tracking_id:
------
## 7 - ok
sed -i 's/tracking_id\s*:.*/tracking_id:/' _config.yml
    tracking_id          : UA-3769691-2
    tracking_id:
------
## 8 - ok
sed -i "" 's/tracking_id\s*:.*/tracking_id:/' _config.yml
    tracking_id          : UA-3769691-2
    tracking_id:

```

The script simulates just the sed command in-place editing piece of script/package. 
Tests 1 - 4 use -e option, and put double quotes around the s/regex/subst/ clause
Tests 5 - 8 don't use -e option, and put single quotes around the s/regex/subst/ clause
Tests 1 and 5 have a space character between -i and 2 single quotes  (gives the original error and the tests are commented out in the script)
Tests 2 and 6 have no space character between -i and 2 single quotes
Tests 3 and 7 supply no extension whatsoever to the -i option
Tests 4 and 8 have a space character between -i and 2 double quotes  (your latest recommendation)

Tests 2-4, and 6-8 all work in the desired fashion.

Probably ... in Tests 4 and 8 the two double quotes are interpreted as being superfluous in the shell, and so the actual command that's run is the same as in Tests 3 and 7.

Test 7 is my preferred option. The -e option isn't required because we are only running a single sed substitution operation.

hope this helps
bazzaar

Reply:
@becca all this sed is making my head spin 🙃 

can we compromise based on your earlier comment (https://github.com/github/training-kit/pull/672#issuecomment-478905267) and make this read  `sed -i "" 's/tracking_id *:.*/tracking_id:/' _config.yml` or `sed -i "" 's/tracking_id.*:.*/tracking_id:/' _config.yml`. Bash and `/usr/bin/sh` as per the file header does not interpret \s appropriately. I've tested this on my machine and 2 other coworkers. These ways work and the `\s` option does not.

Thoughts?

Reply:
Brent, was your comment meant for me? From my point of view, as long as the fix does the job, please feel free to decide how you want to proceed. I've just tried to give you some options, it's the github team's project, when all said and done. bazzaar.

Reply:
@bazzaar (Sorry, autocomplete wasnt working before) Please accept this change 

```suggestion
sed -i "" 's/tracking_id *:.*/tracking_id:/' _config.yml
```

Reply:
change accepted.
bazzaar
(is there a button I need to click?)

Reply:
@bazzaar i think now that I unresolved the conversation, my "suggested change" should be committable by you?